plugin/converter/starlark/starlark.go

```diff
@@ -55,33 +53,7 @@ var (
 	ErrCannotLoad = errors.New("starlark: cannot load external scripts")
 )
 
-// New returns a conversion service that converts the
-// starlark file to a yaml file.
-func New(enabled bool) core.ConvertService {
-	return &starlarkPlugin{
-		enabled: enabled,
-	}
-}
-
-type starlarkPlugin struct {
-	enabled bool
-}
-
-func (p *starlarkPlugin) Convert(ctx context.Context, req *core.ConvertArgs) (*core.Config, error) {
-	if p.enabled == false {
-		return nil, nil
-	}
-
-	// if the file extension is not jsonnet we can
-	// skip this plugin by returning zero values.
-	switch {
-	case strings.HasSuffix(req.Repo.Config, ".script"):
-	case strings.HasSuffix(req.Repo.Config, ".star"):
-	case strings.HasSuffix(req.Repo.Config, ".starlark"):
-	default:
-		return nil, nil
-	}
-
+func Parse(req *core.ConvertArgs, template *core.Template, templateData map[string]interface{}) (file *string, err error) {
```

Comment:
returning a pointer to a string in this scenario is uncommon.  I would change from `*string` to `string`.  Also I am not sure we need to name the return variables since they are not being referenced anywhere.  I would change this to the following:

```diff
-Parse(req *core.ConvertArgs, template *core.Template, templateData map[string]interface{}) (file *string, err error)
+Parse(req *core.ConvertArgs, template *core.Template, templateData map[string]interface{}) (string, error)
```

Reply:
Good to know. Updated.